neptune/new/internal/backends/utils.py

```diff
@@ -42,11 +44,19 @@
 _logger = logging.getLogger(__name__)
 
 
+MAX_RETRY_TIME = 30
+retries_timeout = os.getenv(NEPTUNE_RETRIES_TIMEOUT_ENV, 60)
+
+
 def with_api_exceptions_handler(func):
 
     def wrapper(*args, **kwargs):
         last_exception = None
-        for retry in range(0, 3):
+        start_time = time.time()
```

Comment:
kinda nitpicking, but [time.monotonic](https://docs.python.org/3/library/time.html#time.monotonic) is much safer (accounts for system clock updates, tz updates etc)